kolibri/core/content/models.py

```diff
@@ -215,6 +232,29 @@ class ContentNode(MPTTModel):
         max_length=1800, blank=True
     )  # for fuzzy search in title and description
     lang = models.ForeignKey("Language", blank=True, null=True)
+    # Fields used only on Kolibri and not imported from a content database
+    # importable is used to indicate within a particular import scenario
+    # whether a contentnode is importable or not - always true when importing from Studio
+    importable = models.NullBooleanField()
+    # For all of these values, annotation is done in a Node local way, so it is specific
+    # to this node within the channel, and also does not dedupe.
+    # If you want true values for the size and number of resources for an entire tree
+    # the channelmetadata has deduped data for the entire channel - but this would be
+    # costly to calculate for every single node.
+    # Total count of importable resources for this node - this will change
+    # depending on import source.
+    importable_resources = models.IntegerField(default=0, null=True, blank=True)
+    # Total importable size for this node - this will change depending on content
+    # source.
+    importable_file_size = models.BigIntegerField(default=0, null=True, blank=True)
+    # Total number of importable coach only resources for this node
+    importable_coach_contents = models.IntegerField(default=0, null=True, blank=True)
+    # Total count of on device resources for this node.
+    on_device_resources = models.IntegerField(default=0, null=True, blank=True)
+    # Total file size on disk for this node
+    on_device_file_size = models.BigIntegerField(default=0, null=True, blank=True)
+    # Total number of coach only resources for this node
+    num_coach_contents = models.IntegerField(default=0, null=True, blank=True)
```

Comment:
Yeah, that's my main misgiving about the current implementation - the reason that I did not go even further in this direction is because the content import process is already problematic with simultaneous users.

However, I think your suggestion is more scalable, because we can keep these multiple distinct annotations around for longer and don't have to rewrite them as much.

The other thing I like about this approach is that it doesn't bloat the ContentNodes with a bunch of unnecessary cruft for their core function. Will have a stab at this refactor within the scope of this PR, so that we can avoid having to add and then remove these fields in future.